src/main/java/org/opentripplanner/transit/raptor/rangeraptor/path/ForwardPathMapper.java

```diff
@@ -99,7 +102,12 @@ private void setRangeRaptorIterationDepartureTime(int iterationDepartureTime) {
             from.stop(),
             departureTime,
             from.arrivalTime(),
+            RaptorCostConverter.toOtpDomainCost(from.cost()),
```

Comment:
What happens, when there are multiple legs generated from on AccessPath?

Reply:
We do not calculate the cost "properly" for the street leg/legs - we threat them as "one walking leg" in Raptor for each access/transfer/egress(Raptor leg). The cost returned on the API is the cost calculated by Raptor, not AStar, hence the simplification above. This is something we want to fix, but until then this PR just assign the entire cost (calculated by Raptor) on the first API leg in the access/transfer/egress leg(which may consist of more than one API Leg). If there is more than one API Leg, the reminding leg-generalized-cost is set to 0(zero).